src/main/java/hudson/remoting/Launcher.java

```diff
@@ -211,20 +221,16 @@ public void setConnectTo(String target) {
      *
      * @param ignored
      *      This is ignored.
+     * @deprecated {use {@link #noCertificateCheck}
      */
-    @Option(name="-noCertificateCheck", forbids = "-cert")
+    @Deprecated
     public void setNoCertificateCheck(boolean ignored) throws NoSuchAlgorithmException, KeyManagementException {
         System.out.println("Skipping HTTPS certificate checks altogether. Note that this is not secure at all.");
 
         this.noCertificateCheck = true;
-        SSLContext context = SSLContext.getInstance("TLS");
-        context.init(null, new TrustManager[]{new NoCheckTrustManager()}, new java.security.SecureRandom());
-        HttpsURLConnection.setDefaultSSLSocketFactory(context.getSocketFactory());
-        // bypass host name check, too.
-        HttpsURLConnection.setDefaultHostnameVerifier(new NoCheckHostnameVerifier());
```

Comment:
This entrypoint set the default SSL socket factory and the default hostname verifier JVM-wide, not just for the HTTP connections made by Remoting. The other entrypoint only set these values for the HTTP connections made by Remoting. Whenever dealing with a matter of security where each entrypoint had different behavior, we always chose the more secure option. Thus, we prefer the behavior of the other entrypoint which does not change JVM-wide behavior.